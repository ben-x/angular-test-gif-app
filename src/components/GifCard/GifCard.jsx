import React from 'react';

import {GifCardDiv} from './GifCard.styles'

const GifCard = ({data}) => {
console.log("GifCard -> data", data)
  
  return(<GifCardDiv src={`${data?.images?.downsized?.url}`} alt={`${data?.slug}`} />)
};

export default GifCard;
