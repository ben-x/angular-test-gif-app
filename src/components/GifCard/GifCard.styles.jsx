import styled from 'styled-components';

const GifCardDiv = styled.div`
  border-radius:5px;
  padding: 15px;
  border: solid 2px #c2c2c2;
`;

export {GifCardDiv}
