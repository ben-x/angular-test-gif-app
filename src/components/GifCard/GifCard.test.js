import React from "react";
import { shallow } from "enzyme";
import GifCard from "./GifCard";

const props = {
    data: {},
    getGif: jest.fn()
};

const wrap = shallow(<GifCard {...props} />);

describe("<GifCard />", () => {
    it("renders the component successfully", () => {
        expect(wrap).toMatchSnapshot();
    });
});
