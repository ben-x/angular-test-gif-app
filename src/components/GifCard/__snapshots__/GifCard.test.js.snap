// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifCard /> renders the component successfully 1`] = `ShallowWrapper {}`;
